src/rustllvm/CoverageMappingWrapper.cpp

```diff
@@ -0,0 +1,127 @@
+#include "rustllvm.h"
+#include "llvm/ProfileData/Coverage/CoverageMapping.h"
+#include "llvm/ProfileData/Coverage/CoverageMappingWriter.h"
+#include "llvm/ProfileData/InstrProf.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include "llvm/ADT/ArrayRef.h"
+
+#include <iostream>
+
+using namespace llvm;
+
+extern "C" SmallVectorTemplateBase<coverage::CounterExpression>
```

Comment:
Just a general note, including "SmallVector" in the name strikes me as exposing unnecessary implementation details in an API. But maybe there is precedent for this elsewhere in the rustllvm bindings, I don't know.

Reply:
I had the same gut reaction, and I'm open to changing it, but the reason I did it this way is because there is a constraint. Each of these `SmallVector<>`s are eventually wrapped in an `ArrayRef<>` (via `makeArrayRef` in `llvm::ADT::ArrayRef`), and while there are other vector classes that may also work, the set of "owner" vector types is constrained by the "make..." statics. (I used SmallVector since that's what clang used also.)